artifactory/commands/utils/npmcmdutils.go

```diff
@@ -255,3 +257,46 @@ func PrintMissingDependencies(missingDependencies []buildinfo.Dependency) {
 		"Make sure the dependencies are available in Artifactory for this build.\n" +
 		"Deleting the local cache will force populating Artifactory with these dependencies.")
 }
+
+func BackupFile(filePath, backupPath string) (restore func() error, err error) {
```

Comment:
Let's add a description for this function, specifically explaining the returned value and how it should be used. The same goes for the ```createRestoreFileFunc``` function used by this function. The entire code flow is not trivial and requires some explaining to make the reading easier.
I'd be happy to review this again after the descriptions are added, to see if it's easier for me to understand the code flow.

I also suggest we unit test this function.